plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -503,28 +505,28 @@ private static DeltaLakeColumnMetadata mapColumn(TypeManager typeManager, JsonNo
         String physicalName;
         Type physicalColumnType;
         JsonNode metadata = node.get("metadata");
-        if (metadata.has("delta.typeChanges")) {
-            metadata.get("delta.typeChanges").elements().forEachRemaining(DeltaLakeSchemaSupport::verifyTypeChange);
-        }
         switch (mappingMode) {
             case ID:
                 String columnMappingId = metadata.get("delta.columnMapping.id").asText();
                 verify(!isNullOrEmpty(columnMappingId), "id is null or empty");
                 fieldId = OptionalInt.of(Integer.parseInt(columnMappingId));
                 // Databricks stores column statistics with physical name
                 physicalName = metadata.get("delta.columnMapping.physicalName").asText();
-                verify(!isNullOrEmpty(physicalName), "physicalName is null or empty");
-                physicalColumnType = buildType(typeManager, typeNode, true);
                 break;
             case NAME:
                 physicalName = metadata.get("delta.columnMapping.physicalName").asText();
-                verify(!isNullOrEmpty(physicalName), "physicalName is null or empty");
-                physicalColumnType = buildType(typeManager, typeNode, true);
                 break;
             default:
                 physicalName = fieldName;
-                physicalColumnType = columnType;
         }
+        verifyTypeChanges(metadata, typeNode, partitionColumns.contains(physicalName));
```

Comment:
> partitionColumns.contains(physicalName)

This condition seems wrong. `partitionColumns` holds logical column names. You can check with the following steps:
```sh
testing/bin/ptl env up --environment EnvSinglenodeDeltaLakeOss

docker exec -it ptl-spark spark-sql

CREATE TABLE test_part(a int, b int) USING delta location 's3://test-bucket/test_part' PARTITIONED BY (b) TBLPROPERTIES ('delta.columnMapping.mode' = 'name');

Open http://localhost:9080/ (minio-access-key & minio-secret-key)
```

Probably, we could change the argument to `fieldName`, move the verifyTypeChanges call before `switch (mappingMode)` and revert physicalColumnType change:
```java
        verifyTypeChanges(metadata, typeNode, partitionColumns.contains(fieldName));
        switch (mappingMode) {
...
```